src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Storage/FileSystemObsoletePluginsRemover.cs

```diff
@@ -137,5 +150,35 @@ private Task<bool> TryDeleteDirectory(string dir, CancellationToken cancellation
             }, cancellationToken);
         }
 
+        private bool TryDeleteContentsAtomic(string directory)
+        {
+            List<string> files = Directory.EnumerateFiles(directory, "*", SearchOption.AllDirectories).ToList();
+
+            List<FileStream> streams = files
+                .Select(f =>
+                {
+                    try
+                    {
+                        return File.Open(f, FileMode.Open, FileAccess.ReadWrite, FileShare.Delete);
+                    }
+                    catch (IOException)
```

Comment:
should we just catch any exception here?

Reply:
Sure